components/memory/component.go

```diff
@@ -148,62 +116,50 @@ func (c *component) CheckOnce() {
 	}()
 
 	cctx, ccancel := context.WithTimeout(c.ctx, 5*time.Second)
-	vm, err := mem.VirtualMemoryWithContext(cctx)
+	vm, err := c.getVirtualMemoryFunc(cctx)
 	ccancel()
 	if err != nil {
+		log.Logger.Errorw("failed to get virtual memory", "error", err)
 		d.err = err
+		d.healthy = false
+		d.reason = fmt.Sprintf("failed to get virtual memory: %s", err)
 		return
 	}
-
 	d.TotalBytes = vm.Total
 	d.AvailableBytes = vm.Available
 	d.UsedBytes = vm.Used
+	d.UsedPercent = fmt.Sprintf("%.2f", vm.UsedPercent)
 	d.FreeBytes = vm.Free
 	d.VMAllocTotalBytes = vm.VmallocTotal
 	d.VMAllocUsedBytes = vm.VmallocUsed
 
-	cctx, ccancel = context.WithTimeout(c.ctx, 5*time.Second)
-	err = metrics.SetTotalBytes(cctx, float64(vm.Total), d.ts)
-	ccancel()
-	if err != nil {
-		d.err = err
-		return
-	}
-
-	metrics.SetAvailableBytes(float64(vm.Available))
-
-	cctx, ccancel = context.WithTimeout(c.ctx, 5*time.Second)
-	err = metrics.SetUsedBytes(cctx, float64(vm.Used), d.ts)
-	ccancel()
-	if err != nil {
-		d.err = err
-		return
-	}
+	metricTotalBytes.With(prometheus.Labels{}).Set(float64(vm.Total))
+	metricAvailableBytes.With(prometheus.Labels{}).Set(float64(vm.Available))
+	metricUsedBytes.With(prometheus.Labels{}).Set(float64(vm.Used))
+	metricUsedPercent.With(prometheus.Labels{}).Set(vm.UsedPercent)
+	metricFreeBytes.With(prometheus.Labels{}).Set(float64(vm.Free))
 
 	cctx, ccancel = context.WithTimeout(c.ctx, 5*time.Second)
-	err = metrics.SetUsedPercent(cctx, vm.UsedPercent, d.ts)
-	ccancel()
-	if err != nil {
-		d.err = err
-		return
-	}
-
-	metrics.SetFreeBytes(float64(vm.Free))
-
-	cctx, ccancel = context.WithTimeout(c.ctx, 5*time.Second)
-	bpfJITBufferBytes, err := getCurrentBPFJITBufferBytes(cctx)
+	bpfJITBufferBytes, err := c.getCurrentBPFJITBufferBytesFunc(cctx)
```

Comment:
What is bpfJITBufferBytes used for?

Reply:
@eahydra Some contexts https://github.com/awslabs/amazon-eks-ami/issues/1179 and https://github.com/leptonai/gpud/pull/213